src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -101,20 +101,41 @@ static string DecodeFrom64(string encodedData)
         /// <returns>Subscription addresses that were found for the provided messageTypes</returns>
         public async Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context)
         {
-            var subscribers = new List<Subscriber>();
+            var subscribers = new HashSet<Subscriber>(new SubscriberComparer());
             var table = client.GetTableReference(subscriptionTableName);
 
             foreach (var messageType in messageTypes)
```

Comment:
You can't write a single query as every query is targeted at a partition defined by a partition key. Now, after the fix, we'll do some scans but for sure not all the partitions will be scanned, so then again, we'll save time by not doing a full table scan.
We could run this queries using `Task.WhenAll` though.